File: weather_script.js
const axios = require('axios');
const readline = require('readline');

async function getWeatherData({ city, apiKey }) {
    try {
        const response = await axios.get(`https://api.openweathermap.org/data/2.5/weather?q=${city}&appid=${apiKey}`);
        return response.data;
    } catch (error) {
        console.error('Error:', error.message);
        return error;
    }
}

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

rl.question('Please enter a city (default is Tokyo): ', (inputCity) => {
    const settings = {
        apiKey: '7fff74dea4746c53a552bdf858183d69',  // あなたのAPIキーをここに入力してください
        city: inputCity || 'Tokyo'
    };
    getWeatherData(settings)
        .then(data => {
            if (data instanceof Error) {
                console.error('Failed to get weather data');
            } else {
                console.log(data);
            }
        });
    rl.close();
});

File: weather_script.py
import requests
import json

# グローバル定数
BASE_URL = "https://api.openweathermap.org/data/2.5/weather"
API_KEY = "7fff74dea4746c53a552bdf858183d69"  # あなたのAPIキーをここに入力してください

def get_weather_data(city):
    params = {
        "q": city,
        "appid": API_KEY
    }
    response = requests.get(BASE_URL, params=params)
    return response

def main():
    city = input("Please enter a city (default is Tokyo): ") or "Tokyo"
    response = get_weather_data(city)
    if response.status_code == 200:
        data = response.json()
        print(json.dumps(data, indent=4))  # JSONデータを整形して出力
    else:
        print(f"Error occurred: {response.status_code}")

if __name__ == "__main__":
    main()